Bump to go 1.22 broke CI

### Component(s)

_No response_

### Describe the issue you're reporting

Failed build: https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115

```bash
Run for platform in $(echo $PLATFORMS | tr "," "\n"); do
  for platform in $(echo $PLATFORMS | tr "," "\n"); do
    arch=${platform#*/}
    echo "Building manager for $arch"
    make manager ARCH=$arch
  done
  shell: /usr/bin/bash -e {0}
  env:
    PLATFORMS: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
    OTELCOL_VERSION: 0.96.0
    TARGETALLOCATOR_VERSION: 0.96.0
    OPERATOR_OPAMP_BRIDGE_VERSION: 0.96.0
    AUTO_INSTRUMENTATION_JAVA_VERSION: 1.3[2](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:2).1
    AUTO_INSTRUMENTATION_NODEJS_VERSION: 0.46.0
    AUTO_INSTRUMENTATION_PYTHON_VERSION: 0.44b0
    AUTO_INSTRUMENTATION_DOTNET_VERSION: 1.2.0
    AUTO_INSTRUMENTATION_GO_VERSION: v0.10.1-alpha
    AUTO_INSTRUMENTATION_APACHE_HTTPD_VERSION: 1.0.4
    AUTO_INSTRUMENTATION_NGINX_VERSION: 1.0.4
    VERSION_DATE: 2024-0[3](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:3)-11T1[4](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:4):18:[5](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:5)5Z
    VERSION: 0.9[6](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:6).0
Building manager for amd64
mkdir -p /home/runner/work/opentelemetry-operator/opentelemetry-operator/bin
go: downloading sigs.k8s.io/controller-tools v0.12.0
go: downloading github.com/spf13/cobra v1.[7](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:7).0
go: downloading github.com/gobuffalo/flect v1.0.2
go: downloading k[8](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:8)s.io/apiextensions-apiserver v0.27.1
go: downloading k8s.io/apimachinery v0.27.1
go: downloading golang.org/x/tools v0.8.0
go: downloading github.com/fatih/color v1.15.0
go: downloading k8s.io/api v0.27.1
go: downloading sigs.k8s.io/yaml v1.3.0
go: downloading k8s.io/utils v0.0.0-2023020[9](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:9)194617-a36077c30491
go: downloading k8s.io/klog/v2 v2.90.1
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.2.3
go: downloading github.com/google/gofuzz v1.1.0
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading github.com/mattn/go-isatty v0.0.17
go: downloading github.com/go-logr/logr v1.2.3
go: downloading golang.org/x/sys v0.7.0
go: downloading golang.org/x/net v0.9.0
go: downloading golang.org/x/mod v0.[10](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:10).0
go: downloading golang.org/x/text v0.9.0
/home/runner/work/opentelemetry-operator/opentelemetry-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa0b38f]

goroutine 160 [running]:
go/types.(*Checker).handleBailout(0xc000517400, 0xc000747d40)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/check.go:367 +0x88
panic({0xbc5180?, 0x[12](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:12)b2ca0?})
	/opt/hostedtoolcache/go/1.22.1/x64/src/runtime/panic.go:770 +0x[13](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:13)2
go/types.(*StdSizes).Sizeof(0x0, {0xdc0018, 0x12bb420})
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/sizes.go:228 +0x30f
go/types.(*Config).sizeof(...)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/sizes.go:333
go/types.representableConst.func1({0xdc0018?, 0x12bb420?})
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/const.go:76 +0x9e
go/types.representableConst({0xdc63b0, 0x1287560}, 0xc000517400, 0x12bb420, 0x0)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/const.go:92 +0x192
go/types.(*Checker).arrayLength(0xc000517400, {0xdc46c8, 0xc002d22720?})
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/typexpr.go:510 +0x2d3
go/types.(*Checker).typInternal(0xc000517400, {0xdc2ce8, 0xc002d16870}, 0xc002d62b90)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/typexpr.go:299 +0x49d
go/types.(*Checker).definedType(0xc000517400, {0xdc2ce8, 0xc002d16870}, 0x10?)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/typexpr.go:180 +0x37
go/types.(*Checker).typeDecl(0xc000517400, 0xc002d62b90, 0xc002d[15](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:15)9c0, 0x0)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/decl.go:615 +0x44d
go/types.(*Checker).objDecl(0xc0005[17](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:17)400, {0xdcb9c0, 0xc002d62b90}, 0x0)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/decl.go:197 +0xa7f
go/types.(*Checker).packageObjects(0xc000517400)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/resolver.go:681 +0x425
go/types.(*Checker).checkFiles(0xc000517400, {0xc002755668, 0x3, 0x3})
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/check.go:408 +0x1a5
go/types.(*Checker).Files(...)
	/opt/hostedtoolcache/go/1.22.1/x64/src/go/types/check.go:372
sigs.k8s.io/controller-tools/pkg/loader.(*loader).typeCheck(0xc000255380, 0xc000e1f020)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-tools@v0.12.0/pkg/loader/loader.go:286 +0x36a
sigs.k8s.io/controller-tools/pkg/loader.(*Package).NeedTypesInfo(0xc000e1f020)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-tools@v0.12.0/pkg/loader/loader.go:99 +0x39
sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check(0xc00[18](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:18)8fc20, 0xc000e1f020)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-tools@v0.12.0/pkg/loader/refs.go:268 +0x2b7
sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check.func1(0x49?)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-tools@v0.12.0/pkg/loader/refs.go:262 +0x53
created by sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check in goroutine 54
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-tools@v0.12.0/pkg/loader/refs.go:260 +0x1c5
make: *** [Makefile:[19](https://github.com/open-telemetry/opentelemetry-operator/actions/runs/8234359438/job/22515928115#step:7:19)6: generate] Error 2
Error: Process completed with exit code 2.
```

https://github.com/open-telemetry/opentelemetry-operator/commits/main/

--- a/.github/workflows/continuous-integration.yaml
+++ b/.github/workflows/continuous-integration.yaml
@@ -21,7 +21,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: "~1.22.1"
+        go-version: "~1.21.1"
 
     - name: Cache tools
       uses: actions/cache@v4
@@ -45,7 +45,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: "~1.22.1"
+        go-version: "~1.21.1"
 
     - name: Cache tools
       uses: actions/cache@v4
@@ -79,7 +79,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: "~1.22.1"
+        go-version: "~1.21.1"
 
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v3

--- a/.github/workflows/e2e.yaml
+++ b/.github/workflows/e2e.yaml
@@ -45,7 +45,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: "~1.22.0"
+        go-version: "~1.21.3"
     - name: Cache tools
       uses: actions/cache@v4
       with:

--- a/.github/workflows/publish-images.yaml
+++ b/.github/workflows/publish-images.yaml
@@ -19,7 +19,7 @@ jobs:
 
       - uses: actions/setup-go@v5
         with:
-          go-version: '~1.22.1'
+          go-version: '~1.21.3'
 
       - name: Unshallow 
         run: git fetch --prune --unshallow

--- a/.github/workflows/publish-operator-opamp-bridge.yaml
+++ b/.github/workflows/publish-operator-opamp-bridge.yaml
@@ -23,7 +23,7 @@ jobs:
       
       - uses: actions/setup-go@v5
         with:
-          go-version: '~1.22.1'
+          go-version: '~1.21.3'
 
       # TODO: We're currently not using this. Should we?
       - name: Read version

--- a/.github/workflows/publish-target-allocator.yaml
+++ b/.github/workflows/publish-target-allocator.yaml
@@ -23,7 +23,7 @@ jobs:
 
       - uses: actions/setup-go@v5
         with:
-          go-version: '~1.22.1'
+          go-version: '~1.21.3'
 
       # TODO: We're currently not using this. Should we?
       - name: Read version

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -36,7 +36,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: "~1.22.1"
+        go-version: "~1.21.1"
 
     - name: "generate release resources"
       run: make release-artifacts IMG_PREFIX="ghcr.io/open-telemetry/opentelemetry-operator" VERSION=${DESIRED_VERSION}

--- a/.github/workflows/scorecard.yaml
+++ b/.github/workflows/scorecard.yaml
@@ -25,7 +25,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v5
         with:
-          go-version: "~1.22.1"
+          go-version: "~1.21.1"
 
       - name: Check out code into the Go module directory
         uses: actions/checkout@v4

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/open-telemetry/opentelemetry-operator
 
-go 1.22
+go 1.21
 
 retract v1.51.0
 

